### PR TITLE
Refactor navbar template and styles

### PR DESCRIPTION
**Pull request by vishal423**
- Refactor navbar styles to reduce customization done on top of bootstrap styles and reuse bootstrap variables
- Sync navbar template with bootstrap

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green

-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by vishal423** [comment]
Layout after this change:

image

image

image

**Comment by atomfrede** [comment]
I think its good to stay very close to bootstrap default styles +1. We need that for react too in case the navbar was in line with the angular style.

**Comment by atomfrede** [comment]
@deepu105 What do you think about the style changes, don't want to decide on my own, but I already experienced the pain :) Looking at the bs docs and it just doesn't work because jhipster has made some customizations.

**Comment by deepu105** [comment]
@vishal423 can you post a before and after screenshots, please. I'm tight on time to try it out

File: generators/client/templates/angular/src/main/webapp/app/layouts/navbar/navbar.scss.ejs [hunk 1]
```diff
@@ -17,82 +17,57 @@
  limitations under the License.
 -%>
 
+@import '~bootstrap/scss/functions';
```
[the remark this answers is not in the document]
Comment (vishal423) [reply generators/client/templates/angular/src/main/webapp/app/layouts/navbar/navbar.scss.ejs]: bootstrap `variables` partial internally uses `functions` and without this import it doesn't compile. Also, per my understanding `bootstrap-variables` provides a way to override styles and not to reuse variables declared in bootstrap. Here, I am not overriding any color style and reusing what's provided by default in bootstrap. If someone wants to override color styles, then, they need to declare those overrides in `bootstrap-variables` and they should be implicitly applied here.

[the remark this answers is not in the document]
Comment (deepu105) [reply generators/client/templates/angular/src/main/webapp/app/layouts/navbar/navbar.scss.ejs]: ok

**Pull request merged**